yt_dlp/extractor/digitalconcerthall.py

```diff
@@ -0,0 +1,201 @@
+            msg = 'Unable to login'
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 401:
+                resp = self._parse_json(e.cause.read().decode(), None, fatal=False)
+                if resp:
+                    error = resp.get('extra_info') or resp.get('error_description') or resp.get('error')
+                    if error:
+                        msg += ': ' + error
+            raise ExtractorError('Unable to log in: ' + msg)
+
+    def _real_initialize(self):
+        self._login()
+
+    def _real_extract(self, url):
+        MAX_TITLE_LENGTH = 128
+        language, video_id = re.match(self._VALID_URL, url).groups()
+        if not language:
+            language = 'en'
+        self.debug_out("url: " + url + " video_id: " + video_id + " language: " + language)
+        webpage = self._download_webpage(url, video_id)
+        playlist_title = self._html_search_regex(r'<title>(.+?)</title>', webpage, 'title') or \
+            self._og_search_title(webpage)
+        self.debug_out("playlist_title: " + playlist_title)
+
+        # use the API to get other information about the concert
+        vid_info_dict = self._download_json(
+            'https://api.digitalconcerthall.com/v2/concert/'
+            + video_id, video_id, headers={'Accept': 'application/json',
+                                           'Accept-Language': language})
+        embedded = vid_info_dict.get('_embedded')
+        entries = []
+        for embed_type in embedded:
+            # embed_type should be either 'work' or 'interview'
+            # 'work' will be an array of one or more works
+            for item in embedded.get(embed_type):
+                if embed_type == 'interview':
+                    item['is_interview'] = 1
+                else:
+                    item['is_interview'] = 0
+                stream_href = item.get('_links').get('streams').get('href')
+                self.debug_out("JSON URL: " + 'https:' + stream_href)
+                test_dict = self._download_json('https:' + stream_href, video_id,
+                    headers={'Accept': 'application/json',
+                    'Authorization': 'Bearer ' + self._ACCCESS_TOKEN,
+                    'Accept-Language': language})
+                m3u8_url = test_dict['channel']['vod_mixed_hls_h264_hevc_abr_uhd']['stream'][0]['url']
+                self.debug_out('stream URL: ' + m3u8_url)
+
+                formats = self._extract_m3u8_formats(
+                    m3u8_url, video_id, 'mp4', 'm3u8_native', fatal=False)
+                self._sort_formats(formats)
+
+                if item.get('is_interview') == 1:
+                    title = "Interview - " + item.get('title', "unknown interview title")
+                else:
+                    title = (item.get('name_composer') if item.get('name_composer')
+                            else 'unknown composer') + ' - ' + item.get('title', "unknown title")
+                key = item.get('id')
```

Comment:
Don't try to be smart with titles. The test case should just give `Kirill Petrenko conducts Mendelssohn and Shostakovich` because that is the title in the webpage. The artist and other data should be populated in appropriate fields. Also, dont use `unknown ...`